src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -88,6 +88,24 @@ public AdjustmentRule[] GetAdjustmentRules()
             return (AdjustmentRule[])_adjustmentRules.Clone();
         }
 
+        private string? PopulateDisplayName()
+        {
+            // Keep window's implementation to populate via constructor
+            return null;
+        }
+
+        private string? PopulateStandardDisplayName()
+        {
+            // Keep window's implementation to populate via constructor
+            return null;
+        }
+
+        private string? PopulateDaylightDisplayName()
+        {
+            // Keep window's implementation to populate via constructor
+            return null;
+        }
+
```

Comment:
would these ever be called on Windows? if not, then at least assert or throw exception here to catch any future wrong use of it.

Reply:
It looks like there could be cases where `_displayName` is still `null`, like when using the public APIs `CreateCustomTimeZone`. So if for some reason someone used `TimeZoneInfo myTzi = CreateCustomTimeZone(<id>, <offset>, null, null, null, <adjustmentRules>)` and later invoked `myTzi.DisplayName`, this would be hit.

I think in this case, we should honor that the internal fields have not been set and still return `null`? 
But thats a good point.... I wouldn't want `TimeZoneInfo` objects created by `CreateCustomTimeZone` to have their fields unexpectedly populated (and incur the performance hit) if they intended for them to have a `null` value.

Would the best way to check that be by having some sort of boolean to track whether or not it had been set as `null` on purpose?

Reply:
Now that `CreateCustomTimeZone` sets the names to `string.Empty` if they were `null`, these shouldn't be hit.

Reply:
I added `Debug.Assert(false);` in the window's implementation of these methods.

I modified the primary constructor in `TimeZoneInfo.cs` to have `?? string.Empty` for the internal fields (and removed the modifications from `CreateCustomTimeZone` as they will eventually flow to this constructor.
```
private TimeZoneInfo(
                string id,
                TimeSpan baseUtcOffset,
                string? displayName,
                string? standardDisplayName,
                string? daylightDisplayName,
                AdjustmentRule[]? adjustmentRules,
                bool disableDaylightSavingTime,
                bool hasIanaId = false)
```
In doing so, I believe for windows the only other `TimeZoneInfo` constructor that is of interest is https://github.com/dotnet/runtime/blob/cd9d54165b0de56d3974f9969d3e70412c871c4b/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs#L113, which sets the fields to non null values, so we should not hit these `Populate*` methods on Windows (unless I'm missing something else)

Moreover, this doesn't get used anywhere does it?
https://github.com/dotnet/runtime/blob/cd9d54165b0de56d3974f9969d3e70412c871c4b/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs#L1106-L1117

Reply:
> this doesn't get used anywhere does it?

I recall it can be used by some serialization engines like Data Contract serialization. Please keep it for compat.

Reply:
Ah even though it is a private constructor and isn't referenced in the `TimeZoneInfo.*.cs`?